opm/material/densead/DynamicEvaluation.hpp

```diff
@@ -518,7 +518,7 @@ class Evaluation<ValueT, DynamicSize>
 
 private:
 
-    std::vector<ValueT> data_;
+    FastSmallVector<ValueT, 9> data_;
```

Comment:
9 here is from blackoil solvent case, for the well, EvalWell will have 8 derivatives.

Reply:
I guess this is fine if it becomes a template parameter for `DynamicEvaluation` and is set via a eWoms property in the higher-level code...

Reply:
I am coming back to addressing the comments now.

We agree it is good to make it a template parameter, while DynamicEvaluation will have the third template parameter, also Evaluation, unless we do not make DynamicEvaluation a specialized version of Evaluation. 

Please suggest the elegant way you think we should use to address this?

Reply:
I'm not sure if there is something which qualifies as "elegant". I'd add a third template argument `int staticSize = -1` to `Evaluation` which is ignored this if `numDerivs` is not smaller than 0.

Reply:
I added staticSize to the template parameter, while I have not done it through updating the python code. 

@andlaus , you can take a look to see whether it is done properly.